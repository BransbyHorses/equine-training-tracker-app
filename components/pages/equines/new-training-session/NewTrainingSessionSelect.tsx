import { FormControl, InputLabel, MenuItem, Select } from "@mui/material";

export interface NewTrainingSessionSelectProps {
	categories: any[];
	label: string;
	id: string;
	handleChange: (_e: any) => void;
	value: any;
}

export default function NewTrainingSessionSelect(
	props: NewTrainingSessionSelectProps
) {
	// this check is to handle a "Material-UI: You have provided an out-of-range value" warning
	if (props.categories.length === 0) {
		return (
			<FormControl fullWidth>
				<InputLabel>{props.label}</InputLabel>
				<Select fullWidth label={props.label} value=""></Select>
			</FormControl>
		);
	}

	return (
		<FormControl fullWidth>
			<InputLabel>{props.label}</InputLabel>
			<Select
				id={props.id}
				value={props.value}
				label={props.label}
				onChange={props.handleChange}
			>
				{props.categories.length &&
					props.categories.map((category: any) => {
						return (
							<MenuItem key={category.id} value={category.id}>
								{category.name}
							</MenuItem>
						);
					})}
			</Select>
		</FormControl>
	);
}
