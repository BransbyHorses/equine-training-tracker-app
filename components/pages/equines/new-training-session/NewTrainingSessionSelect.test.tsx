import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import NewTrainingSessionSelect, {
	NewTrainingSessionSelectProps,
} from "./NewTrainingSessionSelect";

describe("NewTrainingSessionSelect", () => {
	it("renders empty select when no options are passed", () => {
		const props: NewTrainingSessionSelectProps = {
			categories: [],
			label: "Test Label",
			id: "test-select",
			handleChange: jest.fn(),
			value: "",
		};
		const tree = renderer.create(<NewTrainingSessionSelect {...props} />);
		expect(tree).toMatchSnapshot();
	});
	it("renders select when options are passed", () => {
		const props: NewTrainingSessionSelectProps = {
			categories: [
				{ id: 1, name: "Option 1" },
				{ id: 2, name: "Option 2" },
				{ id: 3, name: "Option 3" },
				{ id: 4, name: "Option 4" },
			],
			label: "Test Label",
			id: "test-select",
			handleChange: jest.fn(),
			value: "",
		};
		const tree = renderer.create(<NewTrainingSessionSelect {...props} />);
		expect(tree).toMatchSnapshot();
	});
});
