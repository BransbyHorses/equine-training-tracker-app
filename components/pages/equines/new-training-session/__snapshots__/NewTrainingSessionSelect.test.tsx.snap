// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTrainingSessionSelect renders empty select when no options are passed 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    Test Label
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1ftyaf0"
      >
        <span>
          Test Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`NewTrainingSessionSelect renders select when options are passed 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    Test Label
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test-select"
      className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="test-select"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1ftyaf0"
      >
        <span>
          Test Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
